sms_wsj/database/utils.py

```diff
@@ -144,21 +145,39 @@ def scenario_map_fn(
     Args:
         example: Example dictionary.
         snr_range: required for noise generation
-        sync_speech_source: pad and/or cut the source signal to match the
-            length of the observations. Considers the offset.
-        add_speech_reverberation_direct:
-            Calculate the speech_reverberation_direct signal.
+        sync_speech_source: Legacy option. The new convention is, that
+            original_source is the unpadded signal, while speech_source is the
+            padded signal.
+            pad and/or cut the source signal to match the length of the
+            observations. Considers the offset.
+        add_speech_reverberation_early:
+            Calculate the add_speech_reverberation_early signal.
         add_speech_reverberation_tail:
             Calculate the speech_reverberation_tail signal.
+        channel_slice: e.g. None (All channels), [4] (Single channel), ...
+            Warning: Use this only for training. It will change the scale of
+            the data and the added white noise.
+            For the scale the standard deviation is estimated and the generated
+            noise shape changes, hence also the values.
+            With this option you can select the interested channels.
+            All RIRs are used to estimate the time of flight, but only the
+            interested channels are convolved with the original/speech source.
+            This reduces computation time and memory consumption.
 
     Returns:
 
     """
-    h = example['audio_data']['rir']  # Shape (K, D, T)
+    h = example['audio_data']['rir']  # Shape (speaker, channel, sample)
 
     # Estimate start sample first, to make it independent of channel_mode
+    # Calculate one rir_start_sample for each speaker.
```

Comment:
Maybe you should add to the comment, that the rir_start_sample is the estimated time of flight.

Reply:
Good point. I added that